fe/fe-core/src/main/java/org/apache/doris/nereids/stats/StatsCalculator.java

```diff
@@ -299,15 +301,212 @@ public Statistics visitLogicalFilter(LogicalFilter<? extends Plan> filter, Void
+        if (ConnectContext.get() != null && ConnectContext.get().getStatementContext() != null) {
+            ConnectContext.get().getStatementContext().setHasUnknownColStats(true);
+        }
+    }
+
+    private void checkIfUnknownStatsUsedAsKey(StatisticsBuilder builder) {
+        if (ConnectContext.get() != null && ConnectContext.get().getStatementContext() != null) {
+            for (Map.Entry<Expression, ColumnStatistic> entry : builder.getExpressionColumnStatsEntries()) {
+                if (entry.getKey() instanceof SlotReference
+                        && ConnectContext.get().getStatementContext().isKeySlot((SlotReference) entry.getKey())) {
+                    if (entry.getValue().isUnKnown) {
+                        ConnectContext.get().getStatementContext().setHasUnknownColStats(true);
+                        break;
+                    }
+                }
+            }
+        }
+    }
+
+    private Statistics computeOlapScan(LogicalOlapScan olapScan) {
+        OlapTable olapTable = olapScan.getTable();
+
+        if (olapScan.getSelectedIndexId() != olapScan.getTable().getBaseIndexId() || olapTable instanceof MTMV) {
+            // mv is selected, return its estimated stats
+            Optional<Statistics> optStats = cascadesContext.getStatementContext()
+                    .getStatistics(olapScan.getRelationId());
+            if (optStats.isPresent()) {
+                double actualRowCount = olapScan.getTable().getRowCountForNereids();
+                // if estimated mv rowCount is more than actual row count, fall back to base table stats
+                if (actualRowCount > optStats.get().getRowCount()) {
+                    return optStats.get();
+                }
+            }
+        }
+
+        StatisticsBuilder builder = new StatisticsBuilder();
+
+        // for system table or FeUt, use ColumnStatistic.UNKNOWN
+        if (StatisticConstants.isSystemTable(olapTable) || !FeConstants.enableInternalSchemaDb
+                || ConnectContext.get() == null
+                || ConnectContext.get().getSessionVariable().internalSession) {
+            for (Slot slot : olapScan.getOutput()) {
+                builder.putColumnStatistics(slot, ColumnStatistic.UNKNOWN);
+            }
+            setHasUnknownColStatsInStatementContext();
+            builder.setRowCount(olapTable.getRowCountForNereids());
+            return builder.build();
+        }
+
+        // for regression shape test, get row count from columnStats.count
+        if (ConnectContext.get() == null || !ConnectContext.get().getSessionVariable().enableStats) {
+            // get row count from any visible slotReference's colStats
+            double rowCount = 1;
+            for (Slot slot : olapScan.getOutput()) {
+                if (isVisibleSlotReference(slot)) {
+                    ColumnStatistic cache = getColumnStatistic(olapTable, slot.getName(),
+                            olapScan.getSelectedIndexIdForMV());
```

Comment:
`getSelectedIndexIdForMV()` i think this is only used in statistic, so impl it in StatsCalculator.java, not in OlapScan

Reply:
analyze module also need this function